generator/webapp/src/main/java/io/fabric8/maven/generator/webapp/WebAppGenerator.java

```diff
@@ -0,0 +1,190 @@
+    }
+
+    protected Map<String, String> getEnv() {
+        Map<String, String> defaultEnv = new HashMap<>();
+        defaultEnv.put("DEPLOY_DIR", getConfig(Config.deploymentDir));
+        return defaultEnv;
+    }
+
+    @Override
+    public List<ImageConfiguration> customize(List<ImageConfiguration> configs) {
+        ImageConfiguration.Builder imageBuilder = new ImageConfiguration.Builder();
+
+        BuildImageConfiguration.Builder buildBuilder = new BuildImageConfiguration.Builder()
+                .assembly(createAssembly())
+                .from(getFrom())
+                .ports(extractPorts())
+                .cmd(getDockerRunCommand())
+                .env(getEnv());
+        addLatestTagIfSnapshot(buildBuilder);
+        imageBuilder
+                .name(getImageName())
+                .alias(getAlias())
+                .buildConfig(buildBuilder.build());
+        configs.add(imageBuilder.build());
+
+        return configs;
+    }
+
+    protected List<String> extractPorts() {
+        List<String> answer = new ArrayList<>();
+        addPortIfValid(answer, getConfig(Config.webPort));
+        addPortIfValid(answer, getConfig(Config.jolokiaPort));
+        return answer;
+    }
+
+    private void addPortIfValid(List<String> list, String port) {
+        if (Strings.isNotBlank(port)) {
+            list.add(port);
+        }
+    }
+
+    private AssemblyConfiguration createAssembly() {
+        return new AssemblyConfiguration.Builder()
+                .basedir(getDefaultBaseDir())
+                .user(getConfig(Config.user))
+                .descriptorRef("webapp")
+                .build();
+    }
+
+
+    @Override
+    protected String getFrom() {
+        String from = super.getFrom();
+
+        if (from == null) {
+
+            if (isJetty()) {
+                return DOCKER_JETTY_FROM;
+            } else if (isTomcat()) {
+                return DOCKER_TOMCAT_FROM;
+            } else if (isWildFly()) {
+                return DOCKER_WILDFLY_FROM;
+            } else {
+                return DOCKER_TOMCAT_FROM;
+            }
+
+        }
+        return from;
+    }
+
+    private String getDockerRunCommand(){
+
+        String cmd = getConfig(Config.cmd);
+        //since we use jboss/wildfly image the fabric8 deploy-and-run.sh will not work
+        //we also make sure the user has not added custom cmd command to maven config
+        if("/opt/tomcat/bin/deploy-and-run.sh".equals(cmd)) {
+            if (isWildFly()) {
+                cmd = "/opt/jboss/wildfly/bin/standalone.sh";
+            } else if (isJetty()) {
+                cmd = "/opt/jetty/bin/deploy-and-run.sh";
+            }
+        }
+
+        return cmd;
+    }
+
+    private String getDefaultBaseDir() {
+
+        String defaultBaseDir = getConfig(Config.deploymentDir);
+
+        if(isWildFly()
+                && "/deployments".equals(defaultBaseDir)){
```

Comment:
@rhuss  - bit confused here .i understood that when developer sets this config for baseDir manually to "/deployments" then we are forcing the plugin to set to /opt/deployments, which I agree its wrong . when you say you want to check config you mean to do via Maven Model --> f8 plugin --> properties ?  some more details help here..